### ISSUE TITLE
integrate APIs

### ISSUE THREAD

**Issue by slowkow** [JEFworks/ubit2#7]
For a richer user experience, consider integrating some of these APIs:
- [x] mygene.info
- [x] myvariant.info
- [x] UCSC genome browser
- [x] Ensembl API [example (broken, needs fixing)](https://github.com/slowkow/canvasEnsembl)
- [x] Arbitrary VCF files or other tabix-indexed files (can use iobio infrastructure for this)

Consider brainstorming some specific use-cases in this thread.

**Comment by slowkow** [comment]
**myvariant.info**

``` js
function myVariant(query) {
  var data_url = 'http://myvariant.info/v1/query?q=' +
    query + '&fields=dbsnp.hg19,dbsnp.rsid,dbsnp.chrom,dbsnp.alt,dbsnp.ref'

  d3.json(data_url, function(err, json) {
    if (json.hits.length) {
      // Take the first (best) hit.
      var h = json.hits[0]
      var chrom = h.dbsnp.chrom,
          start = h.dbsnp.hg19.start,
          end = h.dbsnp.hg19.end,
          rsid = h.dbsnp.rsid
      // Update the region input.
      var el = document.getElementById('region')
      var cf = d3.format(",")
      el.value = chrom + ':' + cf(start) + '-' + cf(end)
      // Update the plot.
      update({
        chrom: chrom,
        start: start - 50000,
        end: end + 50000,
        pos: start,
        rsid: rsid
      })
    } else {
      d3.select("#loading-text")
        .text("[myvariant.info] No matches for " + query)
    }
  })
}
```

**Comment by slowkow** [comment]
Feel free to take all the code from [this example](http://codepen.io/slowkow/pen/ByxLBE/) that integrates [typeahead.js](https://github.com/twitter/typeahead.js) with [mygene.info](http://mygene.info/):

``` js
var numberWithCommas = function (x) {
    return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
};

var format = function (form, datum) {
    return form.replace(/{([^}]+)}/g, function (match, key) {
        return typeof datum[key] != 'undefined' ? datum[key] : '';
    });
};

var posToRegion = function (pos) {
    if (pos.chr && pos.start && pos.end) {
        return pos.chr + ':' + numberWithCommas(pos.start) + '-' + numberWithCommas(pos.end);
    }
    return '';
};

var entrezGeneLink = function (datum) {
    var form = 'Entrez: ' +
        '<a href="http://www.ncbi.nlm.nih.gov/gene/{entrezgene}">' +
        '{entrezgene}</a>';
    if (datum.entrezgene) {
        return format(form, datum);
    }
    return '';
};

var hgncGeneLink = function (datum) {
    var form = 'HGNC: ' +
        '<a href="http://www.genenames.org/cgi-bin/gene_symbol_report' +
        '?hgnc_id=HGNC:{HGNC}">{HGNC}</a>';
    if (datum.HGNC) {
        return format(form, datum);
    }
    return '';
};

var ucscRegionLink = function (pos) {
    var form = 'UCSC: ' +
        '<a href="https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19' +
        '&position=chr{chr}%3A{start}-{end}">' +
        '{chr}:{start}-{end}</a>';
    if (pos.chr && pos.start && pos.end) {
        return format(form, pos);
    }
    return '';
};
```

**Comment by JEFworks** [comment]
Addressed in commit 7b77c50047ecb8676fca78795b79b46cd16b0429
